test/TestTP.js

```diff
@@ -634,94 +776,16 @@ describe('HappyTokenPool', () => {
-      await happyTokenPoolDeployed
-        .connect(signers[2])
-        .swap(pool_id, verification, signers[2].address, validation, ETH_address_index, exchange_ETH_amount, {
-          value: exchange_ETH_amount,
-        })
-
-      const exchange_tokenB_amount = BigNumber('2e10').toFixed()
-      await approveThenSwapToken(testTokenBDeployed, signers[3], tokenB_address_index, pool_id, exchange_tokenB_amount)
-
-      const exchange_tokenC_amount = BigNumber('1e22').toFixed()
-      await approveThenSwapToken(testTokenCDeployed, signers[4], tokenC_address_index, pool_id, exchange_tokenC_amount)
-
-      await helper.advanceTimeAndBlock(2000 * 1000)
-      await happyTokenPoolDeployed.connect(creator).destruct(pool_id)
-
-      const logs = await ethers.provider.getLogs({
-        address: happyTokenPoolDeployed.address,
-        topics: [ethers.utils.keccak256(ethers.utils.toUtf8Bytes(destruct_success_encode))],
-      })
-      const result = abiCoder.decode(destruct_success_types, logs[0].data)
-
-      expect(result)
-        .to.have.property('id')
-        .that.to.be.eq(pool_id)
-      expect(result)
-        .to.have.property('token_address')
-        .that.to.be.eq(testTokenADeployed.address)
-      expect(result).to.have.property('remaining_tokens')
-      expect(result).to.have.property('exchanged_values')
-
-      const ratioETH = fpp.exchange_ratios[1] / fpp.exchange_ratios[0]
-      const ratioB = fpp.exchange_ratios[3] / fpp.exchange_ratios[2]
-      const ratioC = fpp.exchange_ratios[5] / fpp.exchange_ratios[4]
-      const remaining_tokens =
-        fpp.total_tokens -
-        ratioB * exchange_tokenB_amount -
-        ratioC * exchange_tokenC_amount -
-        ratioETH * exchange_ETH_amount
-      const result_remaining_tokens = Number(result.remaining_tokens)
-
-      expect(result_remaining_tokens).to.be.eq(remaining_tokens)
-      const transfer_amount = new BigNumber('1e26')
-      const tokenB_balance = await testTokenBDeployed.balanceOf(creator.address)
-      expect(BigNumber(tokenB_balance.toString()).toString())
-        .to.be.eq(
-          BigNumber(previous_tokenB_balance.toString())
-            .minus(transfer_amount)
-            .plus(BigNumber(exchange_tokenB_amount))
-            .toString(),
-        )
-        .and.to.be.eq(
-          BigNumber('9e26')
-            .plus(BigNumber('2e10'))
-            .toString(),
-        )
-
-      const tokenC_balance = await testTokenCDeployed.balanceOf(creator.address)
-      expect(BigNumber(tokenC_balance.toString()).toString())
-        .to.be.eq(
-          BigNumber(previous_tokenC_balance.toString())
-            .minus(transfer_amount)
-            .plus(BigNumber(exchange_tokenC_amount))
-            .toString(),
-        )
-        .and.to.be.eq(BigNumber('80001e22').toString())
-    })
-
-    it('Should emit DestructSuccess event and withdraw all tokens', async () => {
```

Comment:
There were two same `Should emit DestructSuccess event and withdraw all token` test suites, remove one.